subsys/net/ip/Kconfig

```diff
@@ -752,6 +752,16 @@ config NET_BUF_DATA_POOL_SIZE
 	  This value tell what is the size of the memory pool where each
 	  network buffer is allocated from.
 
+config NET_BUF_VARIABLE_DEFAULT_FRAG_SIZE
```

Comment:
Yes, I was having something like this in mind (see the PR description), I just wasn't sure if it's acceptable to introduce such an API change to a stable API. Although if we can consider net_pkt API as internal, then I guess there's no concern. I can implement it this way, will try to propose something tomorrow.

Reply:
One option is to leave `net_pkt_get_frag()` as is and introduce a new function that adds the len parameter, and then convert all the calls in zephyr tree to use the new one. If there are external users of the function, they can still use the old one.

Reply:
I'm not sure if I'd like to introduce yet another function just to avoid API change... After thinking about it, the migration for out-of-tree users should be rather straightforward, as they could simply use `CONFIG_NET_BUF_DATA_SIZE` as the length parameter - they're bound to use fixed-sized buffers anyway, as it wouldn't work otherwise. 

I've pushed an update with a new proposal - the API is updated and (hopefully) all of the Zephyr codebase updated to provide the expected fragment length as the parameter. Most of the codebase should now be compatible with both fixed-sized and variable-sized buffers, with exception to some tests and `eth_sam_gmac.c` driver, where I've addded dependency to fixed-size configuration (similar to `eth_dwmac.c`) - simply because the driver was too much dependent on the fixed-sized buffers to sort it out w/o more serious refactoring (which I'd rather avoid w/o having hw to verify changes).